jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -272,13 +284,23 @@ boolean isRPCServer() {
 
     @Override
     public io.opentracing.Span start() {
+      // find preferredParent, it is first childOf
+      Reference preferredParent = references.isEmpty() ? null : references.get(0);
+      for (Reference reference: references) {
+        if (References.CHILD_OF.equals(reference.getType()) &&
+                References.FOLLOWS_FROM.equals(preferredParent.getType())) {
+          preferredParent = reference;
+          break;
+        }
+      }
+
       SpanContext context;
-      if (parent == null) {
+      if (preferredParent == null) {
         context = createNewContext(null);
-      } else if (parent.isDebugIDContainerOnly()) {
-        context = createNewContext(parent.getDebugID());
+      } else if (preferredParent.referencedContext().isDebugIDContainerOnly()) {
```

Comment:
huh, I guess we never implemented the ability to set baggage externally, as in Go client.